R/teal_card.R

```diff
@@ -250,22 +250,25 @@ code_chunk <- function(code, ...) {
 #' Builds a `teal_card` from the code and outputs of a `teal_data`
 #' object, preserving the order of code execution and output display.
 #'
-#' @param data (`qenv`) object.
-#' @return A `teal_card` built from the code and outputs in a `qenv`
-#' object.
+#' @inheritParams eval_code-teal_report
+#' @param x (`list`) object from `qenv@code`.
+#' @return A `teal_card` built from the code and outputs in a `qenv` object.
 #' @keywords internal
-.build_card_from_code <- function(data) {
-  card <- teal_card()
-  for (chunk in data@code) {
-    outs <- if (!is.null(attr(chunk, "outputs"))) {
-      sapply(
-        attr(chunk, "outputs"),
-        function(x) structure(x, class = c("chunk_output", class(x))),
-        USE.NAMES = FALSE,
-        simplify = FALSE
+.code_to_card <- function(x, code_block_opts = list()) {
+  elems <- Reduce(
+    function(items, code_elem) {
+      this_chunk <- do.call(code_chunk, c(list(code = code_elem), code_block_opts))
+      this_outs <- Filter( # intentionally remove warnings,messages from the generated report
+        function(x) !inherits(x, "condition"),
+        lapply(
+          attr(code_elem, "outputs"),
+          function(x) if (isS4(x)) x else structure(x, class = c("chunk_output", class(x)))
+        )
```

Comment:
Given we are removing `chunk_outputs` in `eval_code.teal_report` it doesn't make sense to have it here for consistency sake.

```suggestion
        attr(code_elem, "outputs"),
```

Reply:
Some tests should be deleted (searching for `chunk_output`)

Reply:
I don't understand. We still want to flag outputs as `chunk_output`. 
I don't understand `isS4(x)`  - why not to treat S4 objects as chunk outputs? I would remove this condition